crates/bin/pcli/src/command/tx.rs

```diff
@@ -858,13 +867,72 @@ impl TxCmd {
                     .full_viewing_key
                     .ephemeral_address(OsRng, AddressIndex::from(*source));
 
+                let timeout_height = match timeout_height {
+                    Some(h) => h.clone(),
+                    None => {
+                        // look up the height for the counterparty and add 2 days of block time
+                        // (assuming 10 seconds per block) to it
+
+                        // look up the client state from the channel by looking up channel id -> connection id -> client state
+                        let mut ibc_channel_client =
+                            IbcChannelQueryClient::new(app.pd_channel().await?);
+
+                        let req = QueryChannelRequest {
+                            port_id: "transfer".to_string(),
+                            channel_id: format!("channel-{}", channel),
+                        };
+
+                        let channel = ibc_channel_client
+                            .channel(req)
+                            .await?
+                            .into_inner()
+                            .channel
+                            .ok_or_else(|| anyhow::anyhow!("channel not found"))?;
+
+                        let connection_id = channel.connection_hops[0].clone();
+
+                        let mut ibc_connection_client =
+                            IbcConnectionQueryClient::new(app.pd_channel().await?);
+
+                        let req = QueryConnectionRequest {
+                            connection_id: connection_id.clone(),
+                        };
+                        let connection = ibc_connection_client
+                            .connection(req)
+                            .await?
+                            .into_inner()
+                            .connection
+                            .ok_or_else(|| anyhow::anyhow!("connection not found"))?;
+
+                        let mut ibc_client_client =
+                            IbcClientQueryClient::new(app.pd_channel().await?);
+                        let req = QueryClientStateRequest {
+                            client_id: connection.client_id,
+                        };
+                        let client_state = ibc_client_client
+                            .client_state(req)
+                            .await?
+                            .into_inner()
+                            .client_state
+                            .ok_or_else(|| anyhow::anyhow!("client state not found"))?;
+
+                        let tm_client_state = TendermintClientState::try_from(client_state)?;
+
+                        let last_update_height = tm_client_state.latest_height;
+
+                        IbcHeight {
+                            revision_number: last_update_height.revision_number,
+                            revision_height: last_update_height.revision_height + 87400,
```

Comment:
Related to question above: where does `87400` come from? `24 * 60 * 60 = 86400` but the comment at the top of the match suggests it should be _two_ days, not one. Can we var this value and explain its construction in a comment?

Reply:
it should be ((24*60*60)/10) * 2, since we're assuming 10s per block